### ISSUE TITLE
Use of "colgroup" causes floating Thead column width to no longer match table

### ISSUE THREAD

**Issue by savage25** [mkoryak/floatThead#305]
As soon as the header row floats, it adjusts to fixed width.  
I've narrowed the issue down to the use of "colgroup".

I tried using the "getSizingRow" to fix/work-around the issue, but it either isn't working, or I did something wrong.

Here is a jsfiddle of the issue... I tested and recreated the issue in jsfiddle in both IE and Chrome.
https://jsfiddle.net/qte4j6f1/

The jsfiddle includes my attempt at using "getSizingRow".

- browser name/version: Chrome
- OS name: Windows
- jQuery version if not latest: latest
- floatThead version if not latest: I think it's the latest, not really sure how to update in jsfiddle as I used another jsfiddle example you provided as the start of mine

Thanks!

**Comment by mkoryak** [comment]
This plugin uses a colgroup element to match the floated header cell widths with the tbody. If it finds that the table already has a colgroup, it will use that one. 

In your case your colgroup is saying that the table has a single column with colspan 8. 

It will never work unless you remove the colgroup.

What is the point of that colgroup?

**Comment by savage25** [comment]
OK, I thought perhaps that would be the outcome.  
Unfortunately, the blank colgroup tag is being added by a software package that is presenting/hosting the page.  I came up with a work-around, which was to use jquery to remove the colgroup DOM prior to calling floatThead.

Workaround: https://jsfiddle.net/qte4j6f1/1/

Another issue I see is that the horizontal scrolling causes never-ending horizontal-scrolling when the Thead is floated within an auto-overflowed DIV.  Did you notice that, or know why that occurs? I created an even more simplified jsfiddle to demonstrate: https://jsfiddle.net/1vkvpt37/1/ -- if you scroll to the right before scrolling down, the scroll bar hits the right side and stops.  If you then scroll down so the Thead floats, the horizontal scroll-bar now lets you scroll to the right indefinitely and the Thead stays there even though the table contents have scrolled off.

Thanks again -- this tool is great!

**Comment by mkoryak** [comment]
that is because you are using `responsiveContainer` but you should be using `scrollContainer`

https://jsfiddle.net/1vkvpt37/2/

**Comment by savage25** [comment]
Thank you!

**Comment by savage25** [comment]
If I use "responsiveContainer", it keeps the CSS styling of my Thead... if I use "scrollContainer", the floating Thead is transparent and doesn't maintain boarders.
I'll see if I can create a jsfiddle...

**Comment by savage25** [comment]
Actually, you can see it happening with the jsfiddle you updated to: https://jsfiddle.net/1vkvpt37/2/
The border of the floating Thead is transparent -- whereas it is not when using responsiveContainer.

**Comment by mkoryak** [comment]
some styling cannot be matched by the plugin. because the header was ripped out and put into another table. There isn't much I can do about it anyway that would work for every table. 

It is very easy to fix with some css though, for example:

https://jsfiddle.net/1vkvpt37/3/

`border-collapse:separate` makes tables look ancient, at least to me

**Comment by mkoryak** [comment]
in any case, responsiveContainer does something completely different and is not the correct thing to use here, unless you care about the styles more than functionality ;)

**Comment by savage25** [comment]
OK... thanks for your time looking into this and helping me!

**Comment by savage25** [comment]
Thanks again... I "bought you a beer"  :-)

**Comment by mkoryak** [comment]
thank you, much appreciated. ill drink it after work today

**Comment by fengshics** [comment]
I spent a whole day working on this issue and finally I got the solution. I did not specify the width of my original table while the table uses "table-layout:fixed". I found that I must specify the width of the table in order to let the IE(or other browsers) respect the column widths. So I added a css style to my original table as "width:100%" and then it works perfectly on every browser!

Hope it helps.
